joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -91,6 +91,7 @@ trait TypedScope[M <: MethodLike, F <: FieldLike, T <: TypeLike[M, F]](summary:
   def tryResolveTypeReference(typeName: String): Option[T] = {
     typesInScope
       .collectFirst {
+        case typ if typ.name.split("[.]").lastOption.getOrElse("").equals(typeName)       => typ
```

Comment:
`typ.name.split("[.]").lastOption.contains(typeName)` would be more concise (for next time)